skglm/experimental/sqrt_lasso.py

```diff
@@ -0,0 +1,193 @@
+        """
+        self.coef_ = self.path(X, y, alphas=[self.alpha])[1][0]
+        self.intercept_ = 0.  # TODO handle fit_intercept
+        return self
+
+    def path(self, X, y, alphas=None, eps=1e-3, n_alphas=10):
+        """Compute Lasso path.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            Design matrix.
+
+        y : array, shape (n_samples,)
+            Target vector.
+
+        alphas : array, shape (n_alphas,) default None
+            Grid of alpha. If None a path is constructed from
+            (0, alpha_max] with a length ``eps``.
+
+        eps: float, default 1e-2
+            Length of the path. ``eps=1e-3`` means that
+            ``alpha_min = 1e-3 * alpha_max``.
+
+        n_alphas: int, default 10
+            Number of alphas along the path. This argument is
+            ignored if ``alphas`` was provided.
+
+        Returns
+        -------
+        alphas : array, shape (n_alphas,)
+            The alphas along the path where models are computed.
+
+        coefs : array, shape (n_features, n_alphas)
+            Coefficients along the path.
+        """
+        # build path
+        if alphas is None:
+            alpha_max = norm(X.T @ y, ord=np.inf) / (np.sqrt(len(y)) * norm(y))
+            alphas = alpha_max * np.geomspace(1, eps, n_alphas)
+        else:
+            n_alphas = len(alphas)
+            alphas = np.sort(alphas)[::-1]
+
+        n_features = X.shape[1]
+        sqrt_quadratic = compiled_clone(SqrtQuadratic())
+        l1_penalty = compiled_clone(L1(1.))  # alpha is set along the path
+
+        coefs = np.zeros((n_alphas, n_features))
+
+        for i in range(n_alphas):
+            if self.verbose:
+                print(f"======== alpha {i+1} ========")
+
+            l1_penalty.alpha = alphas[i]
+            # no warm start for the first alpha
+            coef_init = coefs[i].copy() if i else np.zeros(n_features)
+
+            try:
+                coef, _, _ = prox_newton(
+                    X, y, sqrt_quadratic, l1_penalty,
+                    w_init=coef_init, max_iter=self.max_iter,
+                    max_pn_iter=self.max_pn_iter,
+                    tol=self.tol, verbose=self.verbose)
+            except Exception as e:
+                # make sure to catch residual error
+                # it's implemented this way as Numba doesn't support custom Exception
+                if not str(e).startswith("SmallResidualException"):
+                    raise
+
+                # save coef despite not converging
+                # coef_init holds a ref to coef
+                coef = coef_init
+                residual = sqrt_quadratic.value(y, coef, X @ coef)
+                warnings.warn(
+                    f"Small residuals will prevent the solver from converging.\n"
+                    f"Consider taking alphas greater than {alphas[i]:.4e}\n"
+                    f"value of residual: {residual:.4e}",
```

Comment:
residual norm rather than value of residual (residuals are a vector) 

+ beware, you have the normalization by n_samples in the current residual. Just do `norm(y - X @ coef)`